### PR TITLE
Add option to define different n_min_threshold dependent on thresholds

### PR DESCRIPTION
**Pull request by JuliaKukulies**
This PR solves #18 by allowing users to input `n_min_threshold` (the minimum number of grid cells that must be above/below the set threshold(s))  as `dict` or `list`. That way, different `n_min_threshold` values are used for different threshold values in the feature detection. 

The code changes encompass: 

- Changes to the function `feature_detection_multithreshold_timestep()` because here, individual threshold values are selected and passed to `feature_detection_threshold()`
- Input validation to make sure that each threshold is assigned to one `n_min_threshold`.  This means that the number of elements of the given `dict` or `list` has to be the same as the number of given thresholds. It is also made sure that these values are sorted accordingly when threshold values are sorted. When `n_min_threshold` is given as `dict`, the keys must correspond to the threshold values. When `n_min_threshold` is given as `list`, it is assumed that each `n_min_threshold` belongs to the value in `threshold` at the same position. 
- Additional test parameters in  `test_feature_detection.py` to test the new feature
- Modifications to the existing notebook on the input parameter `n_min_threshold` to show an example on our RTD page: see `Different n_min_threshold for different threshold values`under `Feature Detection Parameter Examples` https://tobac--208.org.readthedocs.build/en/208/

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [x] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [x] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [x] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Review by snilsn — CHANGES_REQUESTED** [review]
Very good work @JuliaKukulies, a great addition. I found one issue with sorting for the dictionary case and have a suggestion for that.

File: tobac/feature_detection.py [hunk 1]
```diff
@@ -493,13 +493,51 @@ def feature_detection_multithreshold_timestep(
...
+        # check if thresholds in dict correspond to given thresholds
+        if isinstance(n_min_threshold, dict):
+            if threshold_sorted != sorted(
+                n_min_threshold.keys(), reverse=(target == "minimum")
+            ):
+                raise ValueError(
+                    "Ambiguous input for threshold values. If n_min_threshold is given as a dict, the keys not to correspond to the values in threshold."
+                )
+            # sort dictionary by keys (threshold values) so that they match sorted thresholds and get values for n_min_threshold
+            sorted_dict = dict(sorted(n_min_threshold.items()))
```
[the remark this answers is not in the document]
Comment (JuliaKukulies) [reply tobac/feature_detection.py:512]: Thanks for your review and suggestions @snilsn. Agreed, I will change these two things!

File: tobac/feature_detection.py [hunk 2]
```diff
@@ -493,13 +493,51 @@ def feature_detection_multithreshold_timestep(
...
+            n_min_threshold = list(sorted_dict.values())
+        elif isinstance(n_min_threshold, list):
+            # if n_min_threshold is a list, sort it such that it still matches with the sorted threshold values
+            n_min_threshold = [
+                x
+                for _, x in sorted(
+                    zip(threshold, n_min_threshold), reverse=(target == "minimum")
+                )
+            ]
+    elif isinstance(n_min_threshold, np.ndarray):
```
[the remark this answers is not in the document]
Comment (snilsn) [reply tobac/feature_detection.py:522]: I think checking for dict, list or int would be the right way to go. 1d numpy array would work too, but then we would need to test the shape, and also that all entries are ints.

[the remark this answers is not in the document]
Comment (JuliaKukulies) [reply tobac/feature_detection.py:522]: You are right. I kept it as it is and checking for dict, list, and int.

**Review by freemansw1 — APPROVED** [review]
Thanks @JuliaKukulies! I'm happy with this as-is. Feel free to merge when you're ready.

**Comment by JuliaKukulies** [comment]
Thanks @freemansw1 and @snilsn for reviewing this PR!

**Pull request merged**